crates/types/src/services/p2p.rs

```diff
@@ -72,6 +78,108 @@ pub struct GossipData<T> {
 /// Transactions gossiped by peers for inclusion into a block
 pub type TransactionGossipData = GossipData<Transaction>;
 
+/// Transactions that have been confirmed by block producer
+pub type ConfirmationsGossipData = GossipData<PreConfirmationMessage>;
+
+/// A value and an associated signature
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Sealed<Entity> {
+    /// The actual value
+    pub entity: Entity,
+    /// Seal
+    pub signature: Signature,
+}
+
+/// A key that will be used to sign a pre-confirmations
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct DelegatePreConfirmationKey {
+    /// The public key of the person who is allowed to create pre-confirmations.
+    public_key: PublicKey,
+    /// The time at which the key will expire. Used to indicate to the recipient which key
+    /// to use to verify the pre-confirmations--serves the second purpose of being a nonce of
+    /// each key
+    expiration: Tai64,
+}
+
+/// A pre-confirmation is a message that is sent by the block producer to give the _final_
+/// status of a transaction
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Preconfirmation {
+    /// The ID of the transaction that is being pre-confirmed
+    tx_id: TxId,
+    /// The status of the transaction that is being pre-confirmed
+    status: PreconfirmationStatus,
+}
+
+/// Status of a transaction that has been pre-confirmed by block producer
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum PreconfirmationStatus {
+    /// Transaction was squeezed out by the tx pool
+    SqueezedOutByBlockProducer {
+        /// Reason the transaction was squeezed out
+        reason: String,
+    },
+    /// Transaction has been confirmed and will be included in block_height
+    SuccessByBlockProducer {
+        /// The block height at which the transaction will be included
+        block_height: BlockHeight,
+    },
+    /// Transaction will not be included in a block, rejected at `block_height`
+    FailureByBlockProducer {
+        /// The block height at which the transaction will be rejected
+        block_height: BlockHeight,
+    },
+}
+
+/// A collection of pre-confirmations that have been signed by a delegate
+#[derive(Debug, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct Preconfirmations {
+    /// The expiration time of the key used to sign
+    expiration: Tai64,
+    /// The transactions which have been pre-confirmed
+    preconfirmations: Vec<Preconfirmation>,
```

Comment:
What's the idea behind having a vector ? Because we don't have for NewTx and what would be the rule to end a batch of `Preconfirmation` and send it ?

Reply:
We figured it would be more efficient to sign batches of txs, rather than one at a time. That was especially true with KMS signatures, but maybe we can reconsider with the delegate key signing.

Reply:
> Because we don't have for NewTx and what would be the rule to end a batch of Preconfirmation and send it ?
I'm not sure I understand the question completely, but we would probably rely on the executor to return its best values or something. TBD.